const e=JSON.parse(`{"key":"v-74161bcb","path":"/technology/javascript/features/error.html","title":"错误处理机制","lang":"zh-CN","frontmatter":{"title":"错误处理机制","icon":"article","category":"javascript","article":false,"tag":["javascript"],"copyright":"Copyright by 阮一峰","description":"Error 实例对象 JavaScript 解析或运行时，一旦发生错误，引擎就会抛出一个错误对象。JavaScript 原生提供Error构造函数，所有抛出的错误都是这个构造函数的实例。 var err = new Error('出错了'); err.message // \\"出错了\\"","head":[["meta",{"property":"og:url","content":"https://codemake.fun/technology/javascript/features/error.html"}],["meta",{"property":"og:site_name","content":"Yang's Blog"}],["meta",{"property":"og:title","content":"错误处理机制"}],["meta",{"property":"og:description","content":"Error 实例对象 JavaScript 解析或运行时，一旦发生错误，引擎就会抛出一个错误对象。JavaScript 原生提供Error构造函数，所有抛出的错误都是这个构造函数的实例。 var err = new Error('出错了'); err.message // \\"出错了\\""}],["meta",{"property":"og:type","content":"website"}],["meta",{"property":"og:locale","content":"zh-CN"}],["meta",{"property":"og:updated_time","content":"2023-01-11T09:12:15.000Z"}],["meta",{"property":"article:tag","content":"javascript"}],["meta",{"property":"article:modified_time","content":"2023-01-11T09:12:15.000Z"}],["script",{"type":"application/ld+json"},"{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"WebPage\\",\\"name\\":\\"错误处理机制\\",\\"description\\":\\"Error 实例对象 JavaScript 解析或运行时，一旦发生错误，引擎就会抛出一个错误对象。JavaScript 原生提供Error构造函数，所有抛出的错误都是这个构造函数的实例。 var err = new Error('出错了'); err.message // \\\\\\"出错了\\\\\\"\\"}"]]},"headers":[{"level":2,"title":"Error 实例对象","slug":"error-实例对象","link":"#error-实例对象","children":[]},{"level":2,"title":"原生错误类型","slug":"原生错误类型","link":"#原生错误类型","children":[{"level":3,"title":"SyntaxError 对象","slug":"syntaxerror-对象","link":"#syntaxerror-对象","children":[]},{"level":3,"title":"ReferenceError 对象","slug":"referenceerror-对象","link":"#referenceerror-对象","children":[]},{"level":3,"title":"RangeError 对象","slug":"rangeerror-对象","link":"#rangeerror-对象","children":[]},{"level":3,"title":"TypeError 对象","slug":"typeerror-对象","link":"#typeerror-对象","children":[]},{"level":3,"title":"URIError 对象","slug":"urierror-对象","link":"#urierror-对象","children":[]},{"level":3,"title":"EvalError 对象","slug":"evalerror-对象","link":"#evalerror-对象","children":[]},{"level":3,"title":"总结","slug":"总结","link":"#总结","children":[]}]},{"level":2,"title":"自定义错误","slug":"自定义错误","link":"#自定义错误","children":[]},{"level":2,"title":"throw 语句","slug":"throw-语句","link":"#throw-语句","children":[]},{"level":2,"title":"try...catch 结构","slug":"try-catch-结构","link":"#try-catch-结构","children":[]},{"level":2,"title":"finally 代码块","slug":"finally-代码块","link":"#finally-代码块","children":[]},{"level":2,"title":"参考连接","slug":"参考连接","link":"#参考连接","children":[]}],"git":{"createdTime":1673428335000,"updatedTime":1673428335000,"contributors":[{"name":"sunzhenyang","email":"874530670@qq.com","commits":1}]},"readingTime":{"minutes":8.96,"words":2687},"filePathRelative":"technology/javascript/features/error.md","localizedDate":"2023年1月11日","excerpt":"<h2> Error 实例对象</h2>\\n<p>JavaScript 解析或运行时，一旦发生错误，引擎就会抛出一个错误对象。JavaScript 原生提供<code>Error</code>构造函数，所有抛出的错误都是这个构造函数的实例。</p>\\n<div class=\\"language-javascript line-numbers-mode\\" data-ext=\\"js\\"><pre class=\\"language-javascript\\"><code><span class=\\"token keyword\\">var</span> err <span class=\\"token operator\\">=</span> <span class=\\"token keyword\\">new</span> <span class=\\"token class-name\\">Error</span><span class=\\"token punctuation\\">(</span><span class=\\"token string\\">'出错了'</span><span class=\\"token punctuation\\">)</span><span class=\\"token punctuation\\">;</span>\\nerr<span class=\\"token punctuation\\">.</span>message <span class=\\"token comment\\">// \\"出错了\\"</span>\\n</code></pre><div class=\\"line-numbers\\" aria-hidden=\\"true\\"><div class=\\"line-number\\"></div><div class=\\"line-number\\"></div></div></div>","autoDesc":true}`);export{e as data};
